sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/RateLimitedSamplerTests.cs

```diff
@@ -0,0 +1,202 @@
+        var samplingParams = new SamplingParameters(
+            parentContext,
+            traceId,
+            "span",
+            ActivityKind.Internal
+        );
+        System.Threading.Thread.Sleep(100); // Allow for adaptation time
+        var result = sampler.ShouldSample(samplingParams);
+
+        Assert.Equal(SamplingDecision.RecordAndSample, result.Decision);
+    }
+
+    [Fact]
+    public void ReturnsRecordOnlyWhenTracesPerSecondIsZero()
+    {
+        var sampler = new RateLimitedSampler(0);
+        var parentContext = default(ActivityContext); // No parent
+        var traceId = ActivityTraceId.CreateRandom();
+        var samplingParams = new SamplingParameters(
+            parentContext,
+            traceId,
+            "span",
+            ActivityKind.Internal
+        );
+        System.Threading.Thread.Sleep(100); // Allow for adaptation time
+        var result = sampler.ShouldSample(samplingParams);
+
+        Assert.Equal(SamplingDecision.RecordOnly, result.Decision);
+    }
+
+    [Fact]
+    public void AddsSampleRateAttrToSampledInSpan()
+    {
+        var sampler = new RateLimitedSampler(100);
+        // sleeping to allow for the sampler to adapt. If we send the first span before the adaptation time has passed,
+        // the span will not be sampled.
+        System.Threading.Thread.Sleep(100);
+        var parentContext = default(ActivityContext); // No parent
+        var traceId = ActivityTraceId.CreateRandom();
+        var samplingParams = new SamplingParameters(
+            parentContext,
+            traceId,
+            "span",
+            ActivityKind.Internal
+        );
+        var result = sampler.ShouldSample(samplingParams);
+
+        Assert.Contains(result.Attributes, kvp => kvp.Key == "microsoft.sample_rate");
+        var sampleRateAttr = Assert.Single(result.Attributes, kvp => kvp.Key == "microsoft.sample_rate");
+        double sampleRate = (double)sampleRateAttr.Value;
+        Assert.True(sampleRate == 100);
+    }
+
+    [Fact]
+    public void SpansOverRateLimit()
+    {
+        var sampler = new RateLimitedSampler(50);
+        int sampledin = 0;
+        var parentContext = default(ActivityContext);
+
+        System.Threading.Thread.Sleep(100); // Allow for adaptation time
+        // Simulate 1000 spans being created in just over 1 second.
+        for (int i = 0; i < 5; i++)
+        {
+            for (int j = 0; j < 200; j++)
+            {
+                var traceId = ActivityTraceId.CreateRandom();
+                var samplingParams = new SamplingParameters(parentContext, traceId, $"TestActivity_{i}_{j}", ActivityKind.Internal);
+                var result = sampler.ShouldSample(samplingParams);
+                if (result.Decision == SamplingDecision.RecordAndSample)
+                {
+                    sampledin++;
+                }
+            }
+            System.Threading.Thread.Sleep(200); // 0.2 seconds
+        }
```

Comment:
capture time before and after, say the looping takes 1.3 seconds, we would expect 50*1.3